src/components/ec-metroline/components/ec-metroline-item/ec-metroline-item.vue

```diff
@@ -103,69 +103,66 @@
   </div>
 </template>
 
-<script>
-import * as MetrolineItemStatus from '../../enums/metroline-item-status';
-import EcIcon from '../ec-icon';
-
-export default {
-  name: 'EcMetrolineItem',
-  components: { EcIcon },
-  inject: ['metroline'],
-  props: {
-    id: {
-      type: Number,
-      required: true,
-    },
-    badgeText: {
-      type: String,
-    },
-  },
-  computed: {
-    isReadOnly() {
-      return this.metroline.isCompleted;
-    },
-    isLast() {
-      return this.id === this.metroline.lastItemId;
-    },
-    status() {
-      if (this.id < this.metroline.activeItemId || this.metroline.isCompleted) {
-        return MetrolineItemStatus.COMPLETED;
-      }
-
-      if (this.id === this.metroline.activeItemId) {
-        return MetrolineItemStatus.ACTIVE;
-      }
-
-      return MetrolineItemStatus.NEXT;
-    },
-    isNext() {
-      return this.status === MetrolineItemStatus.NEXT;
-    },
-    isActive() {
-      return this.status === MetrolineItemStatus.ACTIVE;
-    },
-    isCompleted() {
-      return this.status === MetrolineItemStatus.COMPLETED;
-    },
-  },
-  created() {
-    this.metroline.register(this.id);
-  },
-  beforeUnmount() {
-    this.metroline.unregister(this.id);
+<script setup>
+import {
+  computed, inject, onBeforeUnmount, onMounted,
+} from 'vue';
+
+import METROLINE_SYMBOL from '../../../../symbols/provide-inject';
```

Comment:
We should keep all constants that belong to some code close to that code. When we use inject in this situation we will use it in components that are in subdirectory of our parent metroline component that provides data, then it is worth placing its symbols inside its directory.

Reply:
So here we could place file constants.js in `ec-metroline` directory.